src/components/Composer.vue

```diff
@@ -311,20 +311,42 @@ export default {
 				this.editorPlainText = val.editorMode === 'plaintext'
 			}
 		},
+		'$route.params.messageUid'(newID) {
+			this.reset()
+		},
 	},
 	beforeMount() {
-		if (this.fromAccount) {
-			this.selectedAlias = this.aliases.find((alias) => alias.id === this.fromAccount)
-		} else {
-			this.selectedAlias = this.aliases[0]
-		}
-
-		this.bodyVal = this.bodyWithSignature(this.selectedAlias, this.bodyVal)
+		this.setAlias()
+		this.initBody()
 	},
 	mounted() {
 		this.$refs.toLabel.$el.focus()
+		// event is triggered when user clicks 'new message' in navigation
+		this.$root.$on('newMessage', () => {
+			this.draftsPromise
+				.then(() => {
+					return this.saveDraft(this.getMessageData)
+				})
+				.then(() => {
+					// wait for the draft to be saved before resetting the message content
```

Comment:
Actually we can do this in the background, right?

Reply:
Not really, as the draft save has to be triggered before resetting the message content. To be more precise, `getMessageData` has to be called before reset sets the message data to ''. As we don't really have control over when the call to `getMessageData` happens, we have to wait for the whole promise to finish (instead of continuing right after the call).

Reply:
You're right. This could be deferred. Fine then :)